### PR TITLE
Remove homebrew

### PR DESCRIPTION
**Pull request by ocefpaf**
Remove all homebrew Software that is pre-installed in the Travis-CI OS X image.

PS: I wonder what sort of issues we will find once we start re-building everything without the software from homebrew.

### CODE REVIEW

**Comment by jakirkham** [comment]
Should add that this fixes https://github.com/conda-forge/conda-smithy/issues/103.

**Comment by jakirkham** [comment]
Should add that we should bump build numbers when deploying this fix to make sure that new binaries are released from the homebrew free environment.

File: conda_smithy/templates/travis.yml.tmpl [hunk 1]
```diff
@@ -27,6 +27,9 @@ env:
 {%- endif %}
 {% endblock %}
 
+before install:
+    - brew remove --force $(brew list)
```
Comment (jakirkham) [inline conda_smithy/templates/travis.yml.tmpl]: Maybe we should do `brew cleanup` to free the space taken up by these packages.

Comment (ocefpaf) [inline conda_smithy/templates/travis.yml.tmpl]: > > @@ -27,6 +27,9 @@ env:
> >  {%- endif %}
> >  {% endblock %}
> >
> > +before install:
> > +    - brew remove --force $(brew list)
>
> Maybe we should do brew cleanup to free the space taken up by these
packages.

Did you ever run out of space while building on the Travis-CI OS X image?

Comment (jakirkham) [inline conda_smithy/templates/travis.yml.tmpl]: No, but it seems like a waste to leave that stuff around if we don't need it or use it.

**Comment by pelson** [comment]
:+1: 

> Should add that we should bump build numbers when deploying this fix to make sure that new binaries are released from the homebrew free environment.

Personally, I think we should do that on a per-case basis, rather than doing it carte blanche.

**Comment by ocefpaf** [comment]
> Personally, I think we should do that on a per-case basis, rather than doing it carte blanche.

Definitely! There will be side effects. One that I just experienced is that this removes `wget`. If a recipe uses that we need to re-install.

**Comment by jakirkham** [comment]
Are you able to use `curl`?

**Comment by ocefpaf** [comment]
> Are you able to use curl?

Probably. It is not in the list :wink: 

https://github.com/conda-forge/gdal-feedstock/pull/24#issuecomment-205073796

**Comment by pelson** [comment]
I have a recipe for curl if we really need it :wink:

**Comment by jakirkham** [comment]
I believe there is a copy in `defaults`. That being said, I think we will want our own copy of `curl` after this PR ( https://github.com/conda-forge/staged-recipes/pull/228 ) is addressed.

File: conda_smithy/templates/travis.yml.tmpl [hunk 2]
```diff
@@ -27,6 +27,9 @@ env:
 {%- endif %}
 {% endblock %}
 
+before install:
```
Comment (jakirkham) [inline conda_smithy/templates/travis.yml.tmpl]: So, this should be `before_install`.

Comment (jakirkham) [inline conda_smithy/templates/travis.yml.tmpl]: See PR ( https://github.com/conda-forge/conda-smithy/pull/121 ) for fix.

**Pull request merged**